lib/types/beacon_chain/beacon_state.ex

```diff
@@ -131,6 +131,19 @@ defmodule Types.BeaconState do
     end)
   end
 
+  def decode_ex(%__MODULE__{} = map) do
+    map
+    |> Map.update!(:validators, &Aja.Vector.new/1)
+    |> Map.update!(:balances, &Aja.Vector.new/1)
+    |> Map.update!(:randao_mixes, &Aja.Vector.new/1)
+    |> Map.update!(:previous_epoch_participation, &Aja.Vector.new/1)
+    |> Map.update!(:current_epoch_participation, &Aja.Vector.new/1)
```

Comment:
Instead of having these here, shouldn't we be able to match against `vec` in the ssz_ex file instead?

Reply:
This is the decoding part. So the result after decoding is not an `Aja.Vector` but a simple `list`.
In this case, for the `beacon_state` a decoded `list` should be converted into an `Aja.Vector`